src/prefect/engine/cloud/flow_runner.py

```diff
@@ -118,8 +119,22 @@ def call_runner_target_handlers(self, old_state: State, new_state: State) -> Sta
         try:
             cloud_state = new_state
             state = self.client.set_flow_run_state(
-                flow_run_id=flow_run_id, version=version, state=cloud_state
+                flow_run_id=flow_run_id,
+                version=version if cloud_state.is_running() else None,
+                state=cloud_state,
             )
+        except VersionLockError as exc:
+            state = self.client.get_flow_run_state(flow_run_id=flow_run_id)
+
+            if state.is_running():
+                raise ENDRUN(state=state)
```

Comment:
we should probably issue a log right before raising "Flow run already in a running state" or something to that effect